crates/data-plane-controller/src/controller.rs

```diff
@@ -162,6 +162,14 @@ impl automations::Executor for Controller {
 }
 
 impl Controller {
+    fn pulumi_secret_envs(&self) -> Vec<(&str, String)> {
+        ["ARM_CLIENT_ID", "ARM_CLIENT_SECRET", "ARM_TENANT_ID", "ARM_SUBSCRIPTION_ID", "VULTR_API_KEY"]
+            .iter()
+            .flat_map(|key|
+                std::env::var(format!("DPC_{key}")).ok().map(|value| (*key, value))
```

Comment:
Please use unwrap_or_default() so that we always overwrite a variable e.x. `ARM_CLIENT_ID`, even if `DPC_ARM_CLIENT_ID` isn't defined.